File: Factorial.cpp
#include<iostream>
using namespace std;

class Factorial {
private:
    int num;
public:
    void setNum(int n) {
        num = n;
    }

    long long calculateFactorial() {
        long long result = 1;
        for(int i = 2; i <= num; i++) {
            result *= i;
        }
        return result;
    }

    void displayFactorial() {
        if(num < 0) {
            cout << "Error! Factorial of a negative number doesn't exist.";
        } else {
            cout << "The factorial of " << num << " is " << calculateFactorial();
        }
    }
};

int main() {
    Factorial factorialObj;
    int num;
    cout << "Enter a positive integer: ";
    cin >> num;
    factorialObj.setNum(num);
    factorialObj.displayFactorial();
    return 0;
}

File: factorial.py
def factorial(n: int) -> int:
    """
    Returns the factorial of n using recursion.
    """
    if n < 0:
        raise ValueError("Factorial is not defined for negative numbers")
    return 1 if n == 0 else n * factorial(n - 1)


if __name__ == "__main__":
    num = int(input("Enter a number: "))
    print("Factorial:", factorial(num))
